pkg/cnab/extended_bundle.go

```diff
@@ -36,6 +40,24 @@ func LoadBundle(c *portercontext.Context, bundleFile string) (ExtendedBundle, er
 	return NewBundle(*bun), nil
 }
 
+func (b ExtendedBundle) Validate() error {
+	err := b.Bundle.Validate()
+	if err != nil {
+		return fmt.Errorf("invalid bundle: %w", err)
+	}
+
+	supported, err := semver.NewConstraint(SupportedVersion)
+	if err != nil {
+		return fmt.Errorf("invalid supported version %s: %w", SupportedVersion, err)
+	}
+	defaultVersion, err := semver.NewVersion(string(bundle.GetDefaultSchemaVersion()))
+	if err != nil {
+		return err
+	}
```

Comment:
I like to move conversions like this into a build time / compiler error so that I don't have to wonder if it's going to work at runtime.

You can define a package variable like so and use a function that panics to ensure that we are always building with a valid value, and then we don't have to check it each time we validate.

```go
var DefaultBundleSchemaVersion = semver.MustParse(string(bundle.GetDefaultSchemaVersion())
```